src/tleextension.c

```diff
@@ -3976,6 +3976,27 @@ _PU_HOOK
 	bool	cmd_done = false;
 	Oid		tleExtensionOid;
 
+	/*
+	 * Explicitly skip TransactionStmts before calling get_extension_oid().
+	 *
+	 * In an aborted transaction, TransactionStmts (e.g. ROLLBACK) will still
+	 * be passed to process utility hooks. However, relcache lookup must be
+	 * done in a transaction state. Because we don't handle TransactionStmts
+	 * in this hook anyway, TransactionStmts can be skipped.
+	 */
+	if (pu_parsetree && IsA(pu_parsetree, TransactionStmt))
+	{
+		if (prev_hook)
+		{
+			_prev_hook;
+		}
+		else
+		{
+			_standard_ProcessUtility;
+		}
```

Comment:
Nonblocking comment, given it's written otherwise further down on the function, but a future refactor could simplify this to:

```suggestion
		if (prev_hook)
			_prev_hook;
		else
			_standard_ProcessUtility;
```